cli/tre/api_client.py

```diff
@@ -189,7 +180,7 @@ def get_auth_token(self, log, scope):
 
         app = msal.PublicClientApplication(
             client_id=self._client_id,
-            authority=f"https://login.microsoftonline.com/{self._aad_tenant_id}",
+            authority=AuthorityBuilder(AZURE_PUBLIC, self._aad_tenant_id),
```

Comment:
But this won't work in gov. Shouldn't we fix that now as well?

Reply:
The goal of this task is to remove the hardcoded URLs,
the next task is about supporting the gov cloud.
the usage of ClientSecretCredential (see code bleow) also requires modifications to support the gov cloud

Reply:
Not sure I understand. I get that this way looks better, but why would we do it if not to support the gov cloud? Why should we split it to 2 PRs and 2 tasks?

Reply:
The rationale was to start with a task without waiting for the decision of how to handle the parameter that controls the cloud type.
It can be done as part of this task, but after the modifications, the work that has to be done is the same as the work that has to be done as part of #3287.